api/wasm/cpp/proxy_wasm_externs.h

```diff
@@ -77,6 +77,22 @@ extern "C" uint32_t proxy_getMetadataStruct(MetadataType type, const char* name_
 extern "C" uint32_t proxy_setMetadataStruct(MetadataType type, const char* name_ptr, size_t name_size,
                                             const char* value_ptr, size_t value_size);
 
+// Generic selectors
+extern "C" uint32_t proxy_resolveSelector0(const char* arg0_ptr, size_t arg0_size,
+                                           const char** value_ptr_ptr, size_t* value_size_ptr);
+extern "C" uint32_t proxy_resolveSelector1(const char* arg0_ptr, size_t arg0_size,
```

Comment:
Why not just take the expression "foo.bar.baz.bit"?  memcpy of short things is relatively cheap.  This also allows more complex expressions.

Reply:
Also, are we sure that 3... ur 4 is the right limit?  I would also call this 4 rather than 3 since that is the number of arguments even if you do use 0 as a base.

Reply:
I count from 0 :) 
I can increase the number of these methods upwards, but I don't think we'll ever need more than 10 or so.

I would prefer avoiding construction of a long selector string for two reasons:
1. It is more friendly towards compilation to WASM. These are symbols that can be pre-allocated in module memory and re-used at runtime.
2. In the future, I want to cover arrays with a null data and the number being the size. This can also help with encoding symbols into a globally indexed table (e.g. assign each symbol an agreed-upon unique number). I don't think it can get more efficient than that on the wire...